### DISCUSSION TITLE
Apply-to-ancestors modifiers

### DISCUSSION THREAD

**Discussion by Soviut** [tailwindlabs/tailwindcss#9693] [Ideas]
There are a few cases where being able to apply a class to all immediate children, or all ancestors, **without** applying to the parent, would be useful. For example, wanting pointer events to work on a parent, but wanting to disable them for all ancestor elements.

Perhaps something like using `>` and `*` prefixes. They could also be suffixes, but the idea here is to mimic the `!` important modifier.

Child ancestors

```html
<div class=">bg-red-500">
  <div>child: I am red
    <div>grandchild: I am not red
```

All ancestors

```html
<div class="*bg-red-500">
  <div>child: I am red
    <div>grandchild: I am red
```

I understand the desire to keep classes as close to the element they affect as possible. However, we have exceptions already like `space-x` and `divide-y`.

**Comment by Carlsia** [comment]
I'm happy to see the issue is soon regulated. Thankful to you for supporting us with the outcome.Thankful for the little by little useful exercise. Has conclusively the ordinary impact!I figure you should check out support site. [my Loyola Chart
](https://www.my-loyola.com/)

**Comment by simonhamp** [comment]
@Soviut @Carlsia Have you seen https://github.com/SamGoody/tailwind-children ?

Might that be similar to what you're after?

**Reply by simonhamp** [reply]
Also, this seems similar to https://github.com/tailwindlabs/tailwindcss/discussions/9413 - see Adam's response there

**Comment by simonhamp** [comment]
So having played around with this for a while, I think you should be able to do this in your case using [Arbitrary Variants](https://tailwindcss.com/docs/hover-focus-and-other-states#using-arbitrary-variants) without needing an extra plugin (since TW 3.1):

```html
<!-- immediate children only -->
<div class="[&>div]:bg-red-500">

<!-- all grandchildren via a specific element -->
<div class="[&>div>*]:bg-red-500">

<!-- all descendants -->
<div class="[&_*]:bg-red-500">
```

Basically when your class looks like `[&_selector]:bg-red-500`, the TW compiler will rewrite it to something like `.\[\&_selector\]\:bg-red-500 selector { background-color: rgb(239 68 68 / var(--tw-bg-opacity)) }`. Pretty powerful and quite flexible.

(NB: Use underscore to represent in the arbitrary variant definition where you would use a space in the typical CSS selector definition, i.e. `div a` would become `[&_div_a]`)

**Reply by Soviut** [reply]
Yeah, I've been looking into arbitrary variants for a while and they definitely provide solutions to this. However, I still think there's room to codify a few common ancestor patterns to make them easy to identify when skimming the HTML.
